File: Models/CommandModel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Printgloo.Models
{
    public class CommandModel : INotifyPropertyChanged
    {
        #region Data Members

        private string _StopMotion;
        public string StopMotion
        {
            get { return _StopMotion; }
            set
            {
                _StopMotion = value;
                OnPropertyChanged(nameof(StopMotion));
            }
        }

        #region TempCommands

        private string _Temp1On;
        public string Temp1On
        {
            get { return _Temp1On; }
            set
            {
                _Temp1On = value;
                OnPropertyChanged(nameof(Temp1On));
            }
        }

        private string _Temp2On;
        public string Temp2On
        {
            get { return _Temp2On; }
            set
            {
                _Temp2On = value;
                OnPropertyChanged(nameof(Temp2On));
            }
        }

        private string _Temp3On;
        public string Temp3On
        {
            get { return _Temp3On; }
            set
            {
                _Temp3On = value;
                OnPropertyChanged(nameof(Temp3On));
            }
        }

        private string _Temp4On;
        public string Temp4On
        {
            get { return _Temp4On; }
            set
            {
                _Temp4On = value;
                OnPropertyChanged(nameof(Temp4On));
            }
        }

        private string _Temp1Off;
        public string Temp1Off
        {
            get { return _Temp1Off; }
            set
            {
                _Temp1Off = value;
                OnPropertyChanged(nameof(Temp1Off));
            }
        }

        private string _Temp2Off;
        public string Temp2Off
        {
            get { return _Temp2Off; }
            set
            {
                _Temp2Off = value;
                OnPropertyChanged(nameof(Temp2Off));
            }
        }

        private string _Temp3Off;
        public string Temp3Off
        {
            get { return _Temp3Off; }
            set
            {
                _Temp3Off = value;
                OnPropertyChanged(nameof(Temp3Off));
            }
        }

        private string _Temp4Off;
        public string Temp4Off
        {
            get { return _Temp4Off; }
            set
            {
                _Temp4Off = value;
                OnPropertyChanged(nameof(Temp4Off));
            }
        }

        private string _Fan1;
        public string Fan1
        {
            get { return _Fan1; }
            set
            {
                _Fan1 = value;
                OnPropertyChanged(nameof(Fan1));
            }
        }

        private string _Fan2;
        public string Fan2
        {
            get { return _Fan2; }
            set
            {
                _Fan2 = value;
                OnPropertyChanged(nameof(Fan2));
            }
        }

        private string _Fan3;
        public string Fan3
        {
            get { return _Fan3; }
            set
            {
                _Fan3 = value;
                OnPropertyChanged(nameof(Fan3));
            }
        }

        #endregion

        #region Temp Reading

        private string _TempRead;
        public string TempRead
        {
            get { return _TempRead; }
            set
            {
                _TempRead = value;
                OnPropertyChanged(nameof(TempRead));
            }
        }

        private string _Temp1Text;
        public string Temp1Text
        {
            get { return _Temp1Text; }
            set
            {
                _Temp1Text = value;
                OnPropertyChanged(nameof(Temp1Text));
            }
        }

        private string _Temp2Text;
        public string Temp2Text
        {
            get { return _Temp2Text; }
            set
            {
                _Temp2Text = value;
                OnPropertyChanged(nameof(Temp2Text));
            }
        }

        private string _Temp3Text;
        public string Temp3Text
        {
            get { return _Temp3Text; }
            set
            {
                _Temp3Text = value;
                OnPropertyChanged(nameof(Temp3Text));
            }
        }

        private string _Temp4Text;
        public string Temp4Text
        {
            get { return _Temp4Text; }
            set
            {
                _Temp4Text = value;
                OnPropertyChanged(nameof(Temp4Text));
            }
        }

        #endregion

        #region OpMotion Commands

        private string _Auger;
        public string Auger
        {
            get { return _Auger; }
            set
            {
                _Auger = value;
                OnPropertyChanged(nameof(Auger));
            }
        }

        private string _Puller;
        public string Puller
        {
            get { return _Puller; }
            set
            {
                _Puller = value;
                OnPropertyChanged(nameof(Puller));
            }
        }

        private string _Spooler;
        public string Spooler
        {
            get { return _Spooler; }
            set
            {
                _Spooler = value;
                OnPropertyChanged(nameof(Spooler));
            }
        }

        private string _Winder;
        public string Winder
        {
            get { return _Winder; }
            set
            {
                _Winder = value;
                OnPropertyChanged(nameof(Winder));
            }
        }

        private string _WinderSetPos;
        public string WinderSetPos
        {
            get { return _WinderSetPos; }
            set
            {
                _WinderSetPos = value;
                OnPropertyChanged(nameof(WinderSetPos));
            }
        }

        private string _WinderMove;
        public string WinderMove
        {
            get { return _WinderMove; }
            set
            {
                _WinderMove = value;
                OnPropertyChanged(nameof(WinderMove));
            }
        }

        #endregion

        #region Op Commands

        private string _OpMotion;
        public string OpMotion
        {
            get { return _OpMotion; }
            set
            {
                _OpMotion = value;
                OnPropertyChanged(nameof(OpMotion));
            }
        }

        private string _FRead;
        public string FRead
        {
            get { return _FRead; }
            set
            {
                _FRead = value;
                OnPropertyChanged(nameof(FRead));
            }
        }

        private string _OpPreset;
        public string OpPreset
        {
            get { return _OpPreset; }
            set
            {
                _OpPreset = value;
                OnPropertyChanged(nameof(OpPreset));
            }
        }

        #endregion

        #endregion

        #region Functions

        public CommandModel()
        {

        }

        #endregion

        #region Property Changed Notification

        public event PropertyChangedEventHandler PropertyChanged;

        private void OnPropertyChanged(string property)
        {
            if (PropertyChanged != null)
            {
                PropertyChanged(this, new PropertyChangedEventArgs(property));
            }
        }

        #endregion
    }
}

File: Models/CommandModel.py
from __future__ import annotations

from typing import Any, Callable


class CommandModel:
    def __init__(self) -> None:
        self.PropertyChanged: list[Callable[[Any, str], None]] = []

        self._StopMotion: str | None = None
        self._Temp1On: str | None = None
        self._Temp2On: str | None = None
        self._Temp3On: str | None = None
        self._Temp4On: str | None = None
        self._Temp1Off: str | None = None
        self._Temp2Off: str | None = None
        self._Temp3Off: str | None = None
        self._Temp4Off: str | None = None
        self._Fan1: str | None = None
        self._Fan2: str | None = None
        self._Fan3: str | None = None
        self._TempRead: str | None = None
        self._Temp1Text: str | None = None
        self._Temp2Text: str | None = None
        self._Temp3Text: str | None = None
        self._Temp4Text: str | None = None
        self._Auger: str | None = None
        self._Puller: str | None = None
        self._Spooler: str | None = None
        self._Winder: str | None = None
        self._WinderSetPos: str | None = None
        self._WinderMove: str | None = None
        self._OpMotion: str | None = None
        self._FRead: str | None = None
        self._OpPreset: str | None = None

    @property
    def StopMotion(self) -> str | None:
        return self._StopMotion

    @StopMotion.setter
    def StopMotion(self, value: str | None) -> None:
        self._StopMotion = value
        self.OnPropertyChanged("StopMotion")

    @property
    def Temp1On(self) -> str | None:
        return self._Temp1On

    @Temp1On.setter
    def Temp1On(self, value: str | None) -> None:
        self._Temp1On = value
        self.OnPropertyChanged("Temp1On")

    @property
    def Temp2On(self) -> str | None:
        return self._Temp2On

    @Temp2On.setter
    def Temp2On(self, value: str | None) -> None:
        self._Temp2On = value
        self.OnPropertyChanged("Temp2On")

    @property
    def Temp3On(self) -> str | None:
        return self._Temp3On

    @Temp3On.setter
    def Temp3On(self, value: str | None) -> None:
        self._Temp3On = value
        self.OnPropertyChanged("Temp3On")

    @property
    def Temp4On(self) -> str | None:
        return self._Temp4On

    @Temp4On.setter
    def Temp4On(self, value: str | None) -> None:
        self._Temp4On = value
        self.OnPropertyChanged("Temp4On")

    @property
    def Temp1Off(self) -> str | None:
        return self._Temp1Off

    @Temp1Off.setter
    def Temp1Off(self, value: str | None) -> None:
        self._Temp1Off = value
        self.OnPropertyChanged("Temp1Off")

    @property
    def Temp2Off(self) -> str | None:
        return self._Temp2Off

    @Temp2Off.setter
    def Temp2Off(self, value: str | None) -> None:
        self._Temp2Off = value
        self.OnPropertyChanged("Temp2Off")

    @property
    def Temp3Off(self) -> str | None:
        return self._Temp3Off

    @Temp3Off.setter
    def Temp3Off(self, value: str | None) -> None:
        self._Temp3Off = value
        self.OnPropertyChanged("Temp3Off")

    @property
    def Temp4Off(self) -> str | None:
        return self._Temp4Off

    @Temp4Off.setter
    def Temp4Off(self, value: str | None) -> None:
        self._Temp4Off = value
        self.OnPropertyChanged("Temp4Off")

    @property
    def Fan1(self) -> str | None:
        return self._Fan1

    @Fan1.setter
    def Fan1(self, value: str | None) -> None:
        self._Fan1 = value
        self.OnPropertyChanged("Fan1")

    @property
    def Fan2(self) -> str | None:
        return self._Fan2

    @Fan2.setter
    def Fan2(self, value: str | None) -> None:
        self._Fan2 = value
        self.OnPropertyChanged("Fan2")

    @property
    def Fan3(self) -> str | None:
        return self._Fan3

    @Fan3.setter
    def Fan3(self, value: str | None) -> None:
        self._Fan3 = value
        self.OnPropertyChanged("Fan3")

    @property
    def TempRead(self) -> str | None:
        return self._TempRead

    @TempRead.setter
    def TempRead(self, value: str | None) -> None:
        self._TempRead = value
        self.OnPropertyChanged("TempRead")

    @property
    def Temp1Text(self) -> str | None:
        return self._Temp1Text

    @Temp1Text.setter
    def Temp1Text(self, value: str | None) -> None:
        self._Temp1Text = value
        self.OnPropertyChanged("Temp1Text")

    @property
    def Temp2Text(self) -> str | None:
        return self._Temp2Text

    @Temp2Text.setter
    def Temp2Text(self, value: str | None) -> None:
        self._Temp2Text = value
        self.OnPropertyChanged("Temp2Text")

    @property
    def Temp3Text(self) -> str | None:
        return self._Temp3Text

    @Temp3Text.setter
    def Temp3Text(self, value: str | None) -> None:
        self._Temp3Text = value
        self.OnPropertyChanged("Temp3Text")

    @property
    def Temp4Text(self) -> str | None:
        return self._Temp4Text

    @Temp4Text.setter
    def Temp4Text(self, value: str | None) -> None:
        self._Temp4Text = value
        self.OnPropertyChanged("Temp4Text")

    @property
    def Auger(self) -> str | None:
        return self._Auger

    @Auger.setter
    def Auger(self, value: str | None) -> None:
        self._Auger = value
        self.OnPropertyChanged("Auger")

    @property
    def Puller(self) -> str | None:
        return self._Puller

    @Puller.setter
    def Puller(self, value: str | None) -> None:
        self._Puller = value
        self.OnPropertyChanged("Puller")

    @property
    def Spooler(self) -> str | None:
        return self._Spooler

    @Spooler.setter
    def Spooler(self, value: str | None) -> None:
        self._Spooler = value
        self.OnPropertyChanged("Spooler")

    @property
    def Winder(self) -> str | None:
        return self._Winder

    @Winder.setter
    def Winder(self, value: str | None) -> None:
        self._Winder = value
        self.OnPropertyChanged("Winder")

    @property
    def WinderSetPos(self) -> str | None:
        return self._WinderSetPos

    @WinderSetPos.setter
    def WinderSetPos(self, value: str | None) -> None:
        self._WinderSetPos = value
        self.OnPropertyChanged("WinderSetPos")

    @property
    def WinderMove(self) -> str | None:
        return self._WinderMove

    @WinderMove.setter
    def WinderMove(self, value: str | None) -> None:
        self._WinderMove = value
        self.OnPropertyChanged("WinderMove")

    @property
    def OpMotion(self) -> str | None:
        return self._OpMotion

    @OpMotion.setter
    def OpMotion(self, value: str | None) -> None:
        self._OpMotion = value
        self.OnPropertyChanged("OpMotion")

    @property
    def FRead(self) -> str | None:
        return self._FRead

    @FRead.setter
    def FRead(self, value: str | None) -> None:
        self._FRead = value
        self.OnPropertyChanged("FRead")

    @property
    def OpPreset(self) -> str | None:
        return self._OpPreset

    @OpPreset.setter
    def OpPreset(self, value: str | None) -> None:
        self._OpPreset = value
        self.OnPropertyChanged("OpPreset")

    def to_dict(self) -> dict[str, str | None]:
        return {
            "StopMotion": self.StopMotion,
            "Temp1On": self.Temp1On,
            "Temp2On": self.Temp2On,
            "Temp3On": self.Temp3On,
            "Temp4On": self.Temp4On,
            "Temp1Off": self.Temp1Off,
            "Temp2Off": self.Temp2Off,
            "Temp3Off": self.Temp3Off,
            "Temp4Off": self.Temp4Off,
            "Fan1": self.Fan1,
            "Fan2": self.Fan2,
            "Fan3": self.Fan3,
            "TempRead": self.TempRead,
            "Temp1Text": self.Temp1Text,
            "Temp2Text": self.Temp2Text,
            "Temp3Text": self.Temp3Text,
            "Temp4Text": self.Temp4Text,
            "Auger": self.Auger,
            "Puller": self.Puller,
            "Spooler": self.Spooler,
            "Winder": self.Winder,
            "WinderSetPos": self.WinderSetPos,
            "WinderMove": self.WinderMove,
            "OpMotion": self.OpMotion,
            "FRead": self.FRead,
            "OpPreset": self.OpPreset,
        }

    @classmethod
    def from_dict(cls, data: dict[str, Any] | None) -> "CommandModel":
        model = cls()
        if not data:
            return model

        for key in model.to_dict().keys():
            if key in data:
                setattr(model, key, data[key])
        return model

    def OnPropertyChanged(self, property_name: str) -> None:
        for callback in list(self.PropertyChanged):
            callback(self, property_name)
